ballerina/websocket_utils.bal

```diff
@@ -118,43 +99,45 @@ isolated function getSubscriptionResponse(Engine engine, __Schema schema, Contex
     return {errors: [{message: errorMessage}]};
 }
 
-isolated function sendWebSocketResponse(websocket:Caller caller, map<string> & readonly customHeaders, string wsType,
-                                        json payload, string? id = ()) returns websocket:Error? {
-    if customHeaders.hasKey(WS_SUB_PROTOCOL) {
-        string 'type = wsType;
-        if customHeaders.get(WS_SUB_PROTOCOL) == GRAPHQL_WS {
-            if wsType == WS_ERROR || wsType == WS_NEXT {
-                'type = WS_DATA;
-            }
-        }
-        json jsonResponse = id != () ? {'type: 'type, id: id, payload: payload} : {'type: 'type, payload: payload};
-        check caller->writeMessage(jsonResponse);
-    } else {
-        check caller->writeMessage(payload);
+isolated function closeConnection(websocket:Caller caller, SubscriptionError cause, decimal timeout = 5) {
+    string reason = cause.message();
+    int statusCode = cause.detail().code;
+    error? closedConnection = caller->close(statusCode, reason, timeout);
+    if closedConnection is error {
+        error err = error("Failed to close WebSocket connection", closedConnection);
+        log:printError(err.message(), stackTrace = err.stackTrace());
```

Comment:
Can't we use `logError` function here?

Reply:
Yes, fixed with [3f4ef7e](https://github.com/ballerina-platform/module-ballerina-graphql/pull/1170/commits/3f4ef7ebef3f3ff29e0b7e484de1bbd6b069a876)